homeassistant/components/tesla_wall_connector/binary_sensor.py

```diff
@@ -0,0 +1,77 @@
+"""Binary Sensors for Tesla Wall Connector."""
+from dataclasses import dataclass
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_BATTERY_CHARGING,
+    DEVICE_CLASS_PLUG,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.const import ENTITY_CATEGORY_DIAGNOSTIC
+
+from . import (
+    WallConnectorEntity,
+    WallConnectorLambdaValueGetterMixin,
+    prefix_entity_name,
+)
+from .const import DOMAIN, WALLCONNECTOR_DATA_VITALS
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class WallConnectorBinarySensorDescription(
+    BinarySensorEntityDescription, WallConnectorLambdaValueGetterMixin
+):
+    """Binary Sensor entity description."""
+
+
+wall_connector_sensors = [
+    WallConnectorBinarySensorDescription(
+        key="vehicle_connected",
+        name=prefix_entity_name("Vehicle connected"),
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+        value_getter=lambda data: data[WALLCONNECTOR_DATA_VITALS].vehicle_connected,
```

Comment:
We usually use `value_fn` (see `august`)

Reply:
Fixed in adcd6fc